common/djangoapps/entitlements/models.py

```diff
@@ -24,3 +54,65 @@ class CourseEntitlement(TimeStampedModel):
         help_text='The current Course enrollment for this entitlement. If NULL the Learner has not enrolled.'
     )
     order_number = models.CharField(max_length=128, null=True)
+    _policy = models.ForeignKey(CourseEntitlementPolicy, null=True, blank=True)
+
+    @property
+    def expired_at_datetime(self):
+        if not self.expired_at and self.get_days_until_expiration() < 0:
+            self.expired_at = datetime.utcnow()
+            self.save()
+        return self.expired_at
+
+    @expired_at_datetime.setter
+    def expired_at_datetime(self, value):
+        self.expired_at = value
+
+    @property
+    def policy(self):
+        return self._policy or CourseEntitlementPolicy()
+
+    @policy.setter
+    def policy(self, value):
+        self._policy = value
+
+    def get_days_since_created(self):
+        """
+        Returns an integer of number of days since the entitlement has been created
+        """
+        utc = pytz.UTC
+        return (datetime.now(tz=utc) - self.created).days
```

Comment:
Does `self.created` need to be converted to UTC time?

Reply:
self.created is stored in the DB as UTC time.  We can verify this.

Reply:
:+1: out of the DB it's UTC.